sdl-core/src/test/scala/io/smartdatalake/workflow/action/generic/transformer/EncryptColumnsTransformerTest.scala

```diff
@@ -0,0 +1,164 @@
+import org.apache.hadoop.conf.Configuration
+import org.apache.hadoop.fs.{FileSystem, Path}
+
+import java.nio.file.Files
+import java.time.LocalDateTime
+
+class EncryptColumnsTransformerTest extends FunSuite {
+  private val tempDir = Files.createTempDirectory("test")
+
+  val statePath = "target/stateTest/"
+  implicit val filesystem: FileSystem = HdfsUtil.getHadoopFsWithDefaultConf(new Path(statePath))
+
+  test("test column encryption") {
+    val sdlb = new DefaultSmartDataLakeBuilder()
+
+    val config = ConfigFactory.parseString(
+      """
+        |actions = {
+        |   act = {
+        |     type = CopyAction
+        |     inputId = src
+        |     outputId = tgt
+        |     metadata {
+        |       feed = test_run
+        |     }
+        |     transformers = [{
+        |       type = EncryptColumnsTransformer
+        |       encryptColumns = ["c2","c3"]
+        |       keyVariable = "CLEAR#A%D*G-KaPdSgVkYp"
+        |       #keyVariable = "CLEAR#keyblabla"
+        |     }]
+        |   }
+        |}
+        |dataObjects {
+        |  src {
+        |    #id = ~{id}
+        |    type = CsvFileDataObject
+        |    path = "target/raw"
+        |  }
+        |  tgt {
+        |    type = ParquetFileDataObject
+        |    path = "target/column_encrypted"
+        |  }
+        |}
+        |""".stripMargin).resolve
+
+    val globalConfig = GlobalConfig.from(config)
+    implicit val instanceRegistry: InstanceRegistry = ConfigParser.parse(config)
+    implicit val session: SparkSession = sparkSessionBuilder(withHive = true, globalConfig.sparkOptions.getOrElse(Map())).getOrCreate()
+    import session.implicits._
+
+    implicit val actionPipelineContext: ActionPipelineContext = TestUtil.getDefaultActionPipelineContext
+    val sdlConfig = SmartDataLakeBuilderConfig(feedSel = "ids:act")
+
+    val srcDO = instanceRegistry.get[CsvFileDataObject]("src")
+    assert(srcDO != None)
+    val dfSrc = Seq(("testData", "Foo", "ice"), ("bar", "Space", "water"), ("gogo", "Space", "water")).toDF("c1", "c2", "c3")
+    srcDO.writeDataFrame(SparkDataFrame(dfSrc), Seq())(TestUtil.getDefaultActionPipelineContext(sdlb.instanceRegistry))
+
+    // Run SDLB
+    implicit val hadoopConf: Configuration = session.sparkContext.hadoopConfiguration
+    val initialSubFeeds: Seq[SparkSubFeed] = Seq(SparkSubFeed(None, srcDO.id, Seq()))
+    val (subFeeds, stats) = sdlb.exec(sdlConfig, SDLExecutionId.executionId1, runStartTime = LocalDateTime.now, attemptStartTime = LocalDateTime.now, actionsToSkip = Map(), initialSubFeeds = initialSubFeeds, dataObjectsState = Seq(), stateStore = None, stateListeners = Seq(), simulation = false, globalConfig = globalConfig)
+
+    // check result
+    val tgt = instanceRegistry.get[ParquetFileDataObject]("tgt")
+    val dfTgt = tgt.getSparkDataFrame()
+    val colName = dfTgt.columns
+    //assert(colName.toSeq == Seq("testcolumn", "secondcolumn", "thirdcolumn"))

```

Comment:
hmm is this test not working or why is the assert commented out?

Reply:
corrected the assert, thanks.